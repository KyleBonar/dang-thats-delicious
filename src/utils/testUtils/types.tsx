export interface MockLink {
  text: string;
  to: string;
  target: "_blank" | "_self";
}
