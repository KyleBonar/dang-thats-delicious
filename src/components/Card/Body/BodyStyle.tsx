import styled from "../../../theme/styled-components";

export const StyledBody = styled.div`
  padding: 1em;
  font-family: AvenirNextReg;

  h4 {
    font-family: FuturaMedium;
  }
`;
StyledBody.displayName = "div";

export const StyledTextContainer = styled.div`
  padding: 0;
  margin: 0;
`;
StyledTextContainer.displayName = "div";
