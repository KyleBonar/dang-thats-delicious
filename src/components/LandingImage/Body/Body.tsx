import * as React from "react";
import { useRouter } from "next/router";
import { useSelector } from "react-redux";

import { AppState } from "../../../redux/configureStore";
import {
  HeroBody,
  HeroTitle,
  StyledDiv,
  StyledSelect,
  StyledInput,
  StyledButton
} from "./BodyStyle";

export interface IBodyProps {}

const Body: React.FunctionComponent<IBodyProps> = props => {
  const _defaultTitleText = "Find your perfect sauce";

  const [search, setSearch] = React.useState("");
  const [selectedValue, setSelectedValue] = React.useState("all");
  const types = useSelector((store: AppState) => {
    return store.sauces.types;
  });

  // assign router
  const router = useRouter();

  return (
    <HeroBody>
      <HeroTitle>{_defaultTitleText}</HeroTitle>
      <form onSubmit={onSubmit}>
        <StyledDiv>
          <StyledSelect
            id="types"
            options={types}
            selectedValue={selectedValue}
            onSelect={e => setSelectedValue(e.target.value)}
          />
          <StyledInput
            id="search"
            type="text"
            onChange={e => setSearch(e.target.value)}
            placeholder="Search..."
          />
          <StyledButton type="submit">Search</StyledButton>
        </StyledDiv>
      </form>
    </HeroBody>
  );

  async function onSubmit(event: React.FormEvent): Promise<null> {
    event.preventDefault();

    let query = "/sauces?";
    if (search.length > 0) {
      query += `srch=${search}`;
    }

    if (selectedValue.toLowerCase() !== "all") {
      query += `&type=${selectedValue}`;
    }

    // take person to sauces page w/ prefilled search query
    router.push(query);
    return null;
  }
};

export default Body;
