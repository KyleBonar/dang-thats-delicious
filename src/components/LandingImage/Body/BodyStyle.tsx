import styled from "../../../theme/styled-components";
import { Button } from "../../Button/Button";
import Select from "../../Select/Select";
import { TextInput } from "../../TextInput/TextInput";

export const StyledContainer = styled.div`
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  padding: 0 2em;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
`;
StyledContainer.displayName = "div";

export const StyledTitle = styled.h1`
  color: ${x => x.theme.landingHeroTextColor};
  font-family: FuturaMedium;
`;
StyledTitle.displayName = "h1";

export const StyledDiv = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
`;
StyledDiv.displayName = "div";

export const StyledSelect = styled(Select)`
  min-height: 30px;
  border: 0px;
  box-sizing: border-box;

  > div {
    width: auto;

    &:after {
      top: 3px;
    }
  }

  select {
    width: auto;
    border: 0;
  }
`;
StyledSelect.displayName = "Select";

export const StyledInput = styled(TextInput)`
  flex-direction: row;
  align-items: stretch;

  input {
    border: 0px;
    margin-top: 0px;
    margin-bottom: 0px;
    width: 100%;
    min-width: auto;
    max-width: 100%;
    padding: 0.66em;
    box-sizing: border-box;
    border: 0px;
    font-size: 1em;

    @media (min-width: ${props => props.theme.smToMd}) {
      min-width: 20em;
    }
  }
`;
StyledInput.displayName = "TextInput";

export const StyledButton = styled(Button)`
  display: flex;
`;
StyledButton.displayName = "Button";
