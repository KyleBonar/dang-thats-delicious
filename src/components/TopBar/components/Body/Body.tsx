import * as React from "react";
import styled from "../../../../theme/styled-components";

const StyledDiv = styled.div`
  position: absolute;
`;

interface BodyProps {
  children:
    | string
    | Element
    | JSX.Element
    | Array<string | Element | JSX.Element>;
  className?: string;
}

const Body: React.SFC<BodyProps> = props => {
  return (
    <StyledDiv className={props.className} {...props}>
      {props.children}
    </StyledDiv>
  );
};
Body.displayName = "Body";

export default Body;
