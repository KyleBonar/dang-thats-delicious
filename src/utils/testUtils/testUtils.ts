import * as React from "react";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";

import { MockCard } from "./types";
import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import { FlashMessageProps } from "../../components/FlashMessage/FlashMessage";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.string,
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.array_of_words(),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = new Array(casual.integer(1, 15))
    .fill(null)
    .map(fakeSauce);
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured: undefined,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const storeConfig = configureStore([]);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const fakeCard = (): MockCard => ({
  title: casual.string,
  description: casual.string,
  to: casual.url,
  showLink: casual.boolean,
  imageLink: casual.url,
  className: casual.string,
  anchorText: casual.string
});

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(0, MIN_PASSWORD_LENGTH);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword
};
