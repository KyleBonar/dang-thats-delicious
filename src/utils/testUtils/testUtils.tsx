import * as React from "react";
import * as enzyme from "enzyme";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";
import { IErrReturn } from "../Err/Err";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.random_element([undefined, casual.string]),
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.random_element([undefined, casual.array_of_words()]),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const generateFakeSauces = (): ISauce[] =>
  new Array(casual.integer(1, 15)).fill(null).map(fakeSauce);

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = generateFakeSauces();
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  const featured: undefined | string[] = casual.random_element([
    undefined,
    allSlugs.length > 0
      ? allSlugs.filter(slug => {
          if (casual.random < 0.4) return slug;
          return;
        })
      : undefined
  ]);

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const middlewares = [thunk];
  const storeConfig = configureStore(middlewares);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(1, MIN_PASSWORD_LENGTH - 1);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

// Function designed to resemble how a network error would look like.
// This will help the catch(err) only have to handle a single format
const generateErr = (): IErrReturn => {
  return {
    response: {
      data: {
        status: 400,
        msg: casual.string,
        isGood: false
      }
    }
  };
};

// lets us mount up a simple custom hook
const mountReactHook = hook => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.shallow(
      <Component>
        {hookValues => {
          Object.assign(componentHook, hookValues);
          return null;
        }}
      </Component>
    );
  });
  return { componentMount, componentHook };
};

export interface ImountReactHookWithReduxStore {
  componentMount: any;
  componentHook: any;
}
const mountReactHookWithReduxStore = (
  hook,
  store
): ImountReactHookWithReduxStore => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.mount(
      <Provider store={store}>
        <Component>
          {hookValues => {
            Object.assign(componentHook, hookValues);
            return null;
          }}
        </Component>
      </Provider>
    );
  });
  return { componentMount, componentHook };
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword,
  generateInValidPassword,
  generateErr,
  mountReactHook,
  mountReactHookWithReduxStore
};
