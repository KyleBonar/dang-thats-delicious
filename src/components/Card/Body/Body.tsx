import * as React from "react";

import { StyledBody, StyledTextContainer } from "./BodyStyle";

export interface IBodyProps {
  title?: string;
  description?: string;
}

const Body: React.FC<IBodyProps> = props => {
  const defaultText = "Loading ...";
  const { title = defaultText, description = defaultText } = props;

  return (
    <StyledBody>
      <h4>{title}</h4>
      <StyledTextContainer>
        <p>
          {description.length > 60
            ? description.substring(0, 59) + "..."
            : description}
        </p>
      </StyledTextContainer>
    </StyledBody>
  );
};

export default Body;
