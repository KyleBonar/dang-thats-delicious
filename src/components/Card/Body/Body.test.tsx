import * as React from "react";
import * as enzyme from "enzyme";

import Body, { IBodyProps } from "./Body";
import { casual, ITERATION_SIZE } from "../../../utils/testUtils/testUtils";

const fakeBody = (): IBodyProps => ({
  title: casual.random_element([undefined, casual.string]),
  description: casual.random_element([
    undefined,
    casual.string,
    casual.sentences(10)
  ])
});

const mockBodies = new Array(ITERATION_SIZE).fill(null).map(fakeBody);

describe("<Body />", () => {
  const defaultText = "Loading ...";
  const defaultLength = 60;

  it("renders", () => {
    mockBodies.forEach(mockBody => {
      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    mockBodies.forEach(mockBody => {
      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it("renders default title", () => {
    mockBodies.forEach(mockBody => {
      if (mockBody.title) return;

      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper.find("h4").text()).toEqual(defaultText);
    });
  });

  it("renders expected title", () => {
    mockBodies.forEach(mockBody => {
      if (!mockBody.title) return;

      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper.find("h4").text()).toEqual(mockBody.title);
    });
  });

  it("renders default description", () => {
    mockBodies.forEach(mockBody => {
      if (mockBody.description) return;

      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper.find("p").text()).toEqual(defaultText);
    });
  });

  it("renders expected description if short", () => {
    mockBodies.forEach(mockBody => {
      if (!mockBody.description) return;
      if (mockBody.description.length > defaultLength) return;

      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper.find("p").text()).toEqual(mockBody.description);
    });
  });

  it("cuts down descriptions if long", () => {
    mockBodies.forEach(mockBody => {
      if (!mockBody.description) return;
      if (mockBody.description.length < defaultLength) return;

      const wrapper = enzyme.shallow(<Body {...mockBody} />);

      expect(wrapper.find("p").text()).toEqual(
        mockBody.description.substring(0, defaultLength - 1) + "..."
      );
    });
  });
});
