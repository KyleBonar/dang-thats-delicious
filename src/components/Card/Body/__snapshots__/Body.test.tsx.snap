// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> matches snapshot 1`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      fugiat error molestiae laborum est possimus assumenda
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 2`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Itaque voluptas ea ducimus autem sit aspernatur. Quam elige...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 3`] = `
<div>
  <h4>
    officiis qui dignissimos dolore qui voluptatem cumque
  </h4>
  <div>
    <p>
      iste est unde rem similique et maxime
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 4`] = `
<div>
  <h4>
    possimus earum animi quis expedita et optio
  </h4>
  <div>
    <p>
      velit voluptate necessitatibus excepturi et qui sunt
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 5`] = `
<div>
  <h4>
    quia assumenda ut neque saepe est labore
  </h4>
  <div>
    <p>
      Ex architecto ut corrupti voluptatem. Dolores dolor volupta...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 6`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Pariatur voluptates odio rem. Iure quia culpa suscipit tota...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 7`] = `
<div>
  <h4>
    commodi at impedit sint perferendis in dolorem
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 8`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      vero quia voluptatibus consequatur provident qui delectus
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 9`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      provident rerum necessitatibus blanditiis deserunt accusant...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 10`] = `
<div>
  <h4>
    facilis est laboriosam tempora aut cupiditate aut
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 11`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      eaque quis omnis ut praesentium vel earum
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 12`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Voluptatem et autem doloremque earum cum. Perferendis id do...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 13`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 14`] = `
<div>
  <h4>
    voluptas ullam animi excepturi laborum veniam asperiores
  </h4>
  <div>
    <p>
      consectetur animi non reiciendis iure laudantium est
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 15`] = `
<div>
  <h4>
    aut ab laborum sint qui id ipsa
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 16`] = `
<div>
  <h4>
    est et sapiente ipsa aut nostrum repudiandae
  </h4>
  <div>
    <p>
      neque ut dolores qui qui architecto aliquid
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 17`] = `
<div>
  <h4>
    voluptatum eligendi eos magni iste doloribus et
  </h4>
  <div>
    <p>
      adipisci saepe quam quia veritatis qui optio
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 18`] = `
<div>
  <h4>
    tenetur iure quos sed quibusdam voluptatem ducimus
  </h4>
  <div>
    <p>
      vel quia fugit totam quia esse quod
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 19`] = `
<div>
  <h4>
    reprehenderit deleniti nihil ex maiores nihil quisquam
  </h4>
  <div>
    <p>
      Rerum quos consequatur. Odio id id repellat maiores iure fu...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 20`] = `
<div>
  <h4>
    nihil illum ut nobis velit velit perspiciatis
  </h4>
  <div>
    <p>
      Aperiam qui voluptatem rerum et cumque quo. Aut dolore labo...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 21`] = `
<div>
  <h4>
    similique laboriosam molestiae modi expedita quaerat eos
  </h4>
  <div>
    <p>
      itaque ut quis incidunt provident assumenda sit
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 22`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      minima sunt deserunt sit aut et quis
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 23`] = `
<div>
  <h4>
    suscipit fuga non est suscipit dicta ipsum
  </h4>
  <div>
    <p>
      Quia fugiat ducimus quia sit eum possimus. Expedita volupta...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 24`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      iure modi quibusdam odit tempore praesentium quaerat
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 25`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 26`] = `
<div>
  <h4>
    odit et est laudantium rerum quam eum
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 27`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 28`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      vero mollitia tempore delectus eos sunt optio
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 29`] = `
<div>
  <h4>
    laborum voluptatem ut unde quidem molestias repudiandae
  </h4>
  <div>
    <p>
      Nostrum deserunt odit. Voluptates incidunt consequuntur qui...
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 30`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      et amet suscipit sed est corporis est
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 31`] = `
<div>
  <h4>
    Loading ...
  </h4>
  <div>
    <p>
      voluptate omnis ratione sed sit distinctio nobis
    </p>
  </div>
</div>
`;

exports[`<Body /> matches snapshot 32`] = `
<div>
  <h4>
    eum aliquam officiis consequatur est facilis illum
  </h4>
  <div>
    <p>
      Loading ...
    </p>
  </div>
</div>
`;
